File: codeforces/round 341/A.java
//============================================================================
// Name        :
// Author      : hussam
// Version     :
// Copyright   :
// Description :
//============================================================================

import java.util.*;

public class mycode {

	public static void main(String[] args) {

		Scanner cin = new Scanner(System.in);
     long minodd =100000000000L,res=0;
     int n=cin.nextInt();
     for(int i=0;i<n;i++)
     {
    	 int num=cin.nextInt();
    		res+=num;
    	if(num%2!=0&&num<minodd)
    	{
    		minodd=num;
    	}
    	 
     }
     
     if(res%2==0)
    	 System.out.println(res);
     else
    	 System.out.println(res-minodd);
		
	}}

File: codeforces/round 341/A.cpp
//============================================================================
// Name        :
// Author      : hussam
// Version     :
// Copyright   :
// Description :
//============================================================================

#include<bits/stdc++.h>
using namespace std;
int main()
{
	int num;
	long long sod = 0, sev = 0, lodd = 9999999999;
	scanf("%d", &num);
	while (num--)
	{

		int inp;
		scanf("%d", &inp);
		if (inp % 2 == 0)
			sev += inp;
		else
		{
			sod += inp;
			if (inp < lodd)
				lodd = inp;
		}
	}

	if ((sod + sev) % 2 != 0)
		sod -= lodd;
	cout << sev + sod << endl;

	return 0;
}